### ISSUE TITLE
False positive for IDE0051

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#32488]
C# Code:

```csharp
using System;
using System.Reflection;

namespace Test
{
    class TestClass
    {
        private static readonly MethodInfo TesteMethod =
            typeof(TestClass).GetTypeInfo().GetMethod(nameof(Test), BindingFlags.Static | BindingFlags.NonPublic);

        private static DateTime Test()
        {
            return DateTime.Now;
        }
    }
}
```

_This issue has been moved from https://developercommunity.visualstudio.com/content/problem/419930/false-positive-for-ide0051.html
VSTS ticketId: 760860_
_These are the original issue comments:_
(no comments)
_These are the original issue solutions:_
(no solutions)

**Comment by mavasani** [comment]
This is by design - please see https://github.com/dotnet/roslyn/issues/31581#issuecomment-445046915 for details.
